From 5b73389c9eb8e4e1ce1ccb8654b17703176eb009 Mon Sep 17 00:00:00 2001
From: Charlie Sharpsteen <source@sharpsteen.net>
Date: Mon, 6 May 2019 21:53:23 -0700
Subject: [PATCH 6/8] Skip yaml-cpp and boost for debian armhf

Let's take a page out of the SLES 15 book and not build these beefy
dependencies into the runtime for debian-9-armhf and debian-10-armhf.
---
 configs/projects/agent-runtime-6.4.x.rb       |  4 +-
 configs/projects/agent-runtime-master.rb      |  4 +-
 configs/projects/agent-runtime-master.rb.orig | 40 +++++++++++++++++++
 3 files changed, 44 insertions(+), 4 deletions(-)
 create mode 100644 configs/projects/agent-runtime-master.rb.orig

diff --git a/configs/projects/agent-runtime-6.4.x.rb b/configs/projects/agent-runtime-6.4.x.rb
index 4f7b564..9880490 100644
--- a/configs/projects/agent-runtime-6.4.x.rb
+++ b/configs/projects/agent-runtime-6.4.x.rb
@@ -33,6 +33,6 @@ project 'agent-runtime-6.4.x' do |proj|
   proj.component 'rubygem-hiera-eyaml'
   proj.component 'rubygem-httpclient'
   # SLES 15 uses the OS distro versions of boost and yaml-cpp:
-  proj.component 'boost' unless platform.name =~ /sles-15/
-  proj.component 'yaml-cpp' unless platform.name =~ /sles-15/
+  proj.component 'boost' unless platform.name =~ /sles-15|debian-(?:9|10)-armhf/
+  proj.component 'yaml-cpp' unless platform.name =~ /sles-15|debian-(?:9|10)-armhf/
 end
diff --git a/configs/projects/agent-runtime-master.rb b/configs/projects/agent-runtime-master.rb
index bc589e0..5ea4446 100644
--- a/configs/projects/agent-runtime-master.rb
+++ b/configs/projects/agent-runtime-master.rb
@@ -41,6 +41,6 @@ project 'agent-runtime-master' do |proj|
   end
 
   # SLES 15 uses the OS distro versions of boost and yaml-cpp:
-  proj.component 'boost' unless platform.name =~ /sles-15/
-  proj.component 'yaml-cpp' unless platform.name =~ /sles-15/
+  proj.component 'boost' unless platform.name =~ /sles-15|debian-(?:9|10)-armhf/
+  proj.component 'yaml-cpp' unless platform.name =~ /sles-15|debian-(?:9|10)-armhf/
 end
diff --git a/configs/projects/agent-runtime-master.rb.orig b/configs/projects/agent-runtime-master.rb.orig
new file mode 100644
index 0000000..7ffabfd
--- /dev/null
+++ b/configs/projects/agent-runtime-master.rb.orig
@@ -0,0 +1,40 @@
+project 'agent-runtime-master' do |proj|
+  # Set preferred component versions if they differ from defaults:
+  proj.setting :ruby_version, '2.5.7'
+  proj.setting :augeas_version, '1.12.0'
+  proj.setting :openssl_version, platform.name =~ /windowsfips-2012r2/ ? '1.0.2' : '1.1.1'
+
+  ########
+  # Load shared agent settings
+  ########
+
+  instance_eval File.read(File.join(File.dirname(__FILE__), '_shared-agent-settings.rb'))
+
+  ########
+  # Settings specific to the master branch
+  ########
+
+  # Directory for gems shared by puppet and puppetserver
+  proj.setting(:puppet_gem_vendor_dir, File.join(proj.libdir, "ruby", "vendor_gems"))
+
+  ########
+  # Load shared agent components
+  ########
+
+  instance_eval File.read(File.join(File.dirname(__FILE__), '_shared-agent-components.rb'))
+
+  ########
+  # Components specific to the master branch
+  ########
+
+  proj.component 'rubygem-concurrent-ruby'
+  proj.component 'rubygem-multi_json'
+  proj.component 'rubygem-optimist'
+  proj.component 'rubygem-highline'
+  proj.component 'rubygem-hiera-eyaml'
+  proj.component 'rubygem-httpclient'
+  proj.component 'rubygem-thor'
+  # SLES 15 uses the OS distro versions of boost and yaml-cpp:
+  proj.component 'boost' unless platform.name =~ /sles-15/
+  proj.component 'yaml-cpp' unless platform.name =~ /sles-15/
+end
-- 
2.25.0

